internal/provisioner/cluster.go

```diff
@@ -234,6 +258,38 @@ func DeleteCluster(clusterID, s3StateStore string, logger log.FieldLogger) error
 	return nil
 }
 
+// waitForKubernetesReadyness will poll a given kubernetes cluster at a regular
+// interval for it to become ready. If the cluster fails to become ready before
+// the provided timeout then an error will be returned.
+func waitForKubernetesReadyness(dns string, kops *kops.Cmd, timeout time.Duration) error {
+	ready := make(chan bool, 1)
```

Comment:
Since there's no actual message going across the wire here, you can just make this unbuffered and replace the send on line 277 with a `close(ready)` to signal. It makes the intention of the channel as a signal mechanism more clear, vs. the traditional messaging passing mechanism.

(I note a different simplification below, though!)